src/zap.ts

```diff
@@ -0,0 +1,132 @@
+import { LiquidityAddition } from "./addLiquidity";
+import { Asset } from "./asset";
+import { getConstantProductZapParams } from "./constantProductCalculator";
+import { PactSdkError } from "./exceptions";
+import { Pool } from "./pool";
+import { Swap } from "./swap";
+import { TransactionGroup } from "./transactionGroup";
+
+/**
+ * All amounts that should be used in swap and add liquidity transactions.
+ */
+export type ZapParams = {
+  swapDeposited: bigint;
+  swapReceived: bigint;
+  primaryAddLiq: bigint;
+  secondaryAddLiq: bigint;
+};
+
+/**
+ * Zap class represents a zap trade on a particular pool, which allows to exchange single asset for PLP token.
+ *
+ * Zap creates a swap function to get second asset from the pool and then adds liquidity using both of those assets. Users may be left with some leftovers due to rounding and slippage settings.
```

Comment:
`Zap creates a swap function` -> `Zap performs a swap`

Reply:
I would add information that zaps are only for constant product pools. For stableswaps, adding only one asset works out of the box.